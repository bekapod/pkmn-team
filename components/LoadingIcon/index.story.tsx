import { storiesOf } from "@storybook/react";
import React from "react";
import LoadingIcon from ".";

storiesOf("icons/LoadingIcon", module)
  .add("default", (): JSX.Element => <LoadingIcon />)
  .add("spinner", (): JSX.Element => <LoadingIcon spinner />);
