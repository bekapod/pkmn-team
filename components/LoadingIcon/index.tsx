import { rgba } from "polished";
import React from "react";
import styled from "styled-components/macro";
import { pulseFade, spin } from "../../helpers/animations";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";

interface Props {
  spinner?: boolean;
  small?: boolean;
  color?: string;
}

const Styled = styled.div`
  animation: ${pulseFade} 2s linear infinite;
  margin: ${variables.spacing.xl}px 0;
  color: ${variables.colors.grayDark};
  font-family: ${variables.fonts.base};
  font-size: ${variables.fontSizes.md}px;
  font-weight: 700;
  letter-spacing: ${variables.spacing.xxs}px;
  line-height: ${lineHeight("md")};
  text-align: center;
  text-transform: uppercase;
`;

const StyledSpinner = styled.div`
  position: relative;
  border: ${({ small, color }: Props): string =>
    `${small ? "5px" : "10px"} solid ${rgba(
      color || variables.colors.secondary,
      0.15
    )}`};
  border-left: ${({ small, color }: Props): string =>
    `${small ? "5px" : "10px"} solid ${color || variables.colors.secondary}`};
  transform: translateZ(0);
  animation: ${spin} 1s linear infinite;

  &,
  &:after {
    width: ${({ small }: Props): number =>
      small ? variables.spacing.lg : variables.spacing.xl}px;
    height: ${({ small }: Props): number =>
      small ? variables.spacing.lg : variables.spacing.xl}px;
    border-radius: 50%;
  }
`;

const loadingText = "Loading";

const LoadingIcon = ({ spinner, ...props }: Props): JSX.Element =>
  spinner ? (
    <StyledSpinner
      data-testid="loading-spinner"
      role="img"
      aria-label={loadingText}
      {...props}
    />
  ) : (
    <Styled>{loadingText}</Styled>
  );

export default LoadingIcon;
