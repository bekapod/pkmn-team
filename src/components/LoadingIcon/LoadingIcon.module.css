.is-small {
}

.text {
  animation-duration: 2000ms;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  margin: var(--spacing-5) 0;
  color: var(--color-cool-grey-600);
  font-family: var(--font-family);
  font-size: var(--font-size-md);
  font-weight: 700;
  letter-spacing: var(--spacing--1);
  line-height: var(--line-height-md);
  text-align: center;
  text-transform: uppercase;
}

.text :global {
  animation-name: pulseFade;
}

.spinner {
  position: relative;
  border-width: 10px;
  border-style: solid;
  border-color: var(--color-cool-grey-200);
  border-left-color: var(--color-cool-grey-600);
  transform: translateZ(0);
  animation-duration: 1250ms;
  animation-timing-function: linear;
  animation-iteration-count: infinite;

  &,
  &:after {
    width: var(--spacing-8);
    height: var(--spacing-8);
    border-radius: 50%;
  }
}

.spinner :global {
  animation-name: spin;
}

.spinner.is-small {
  border-width: 5px;

  &,
  &::after {
    width: var(--spacing-5);
    height: var(--spacing-5);
  }
}

.is-centered {
  position: absolute;
  top: 0;
  bottom: 0;
  margin: auto;
}
