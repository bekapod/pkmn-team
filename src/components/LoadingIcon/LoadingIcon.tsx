import type { ComponentPropsWithRef, FunctionComponent } from 'react';
import cx from 'classnames';

export type LoadingIconProps = {
  isSpinner?: boolean;
  isSmall?: boolean;
  isCentered?: boolean;
};

const loadingText = 'Loading';

export const LoadingIcon: FunctionComponent<
  ComponentPropsWithRef<'div'> & LoadingIconProps
> = ({ isSpinner, isSmall, isCentered, className, ...props }) =>
  isSpinner ? (
    <div
      className={cx(
        'spinner',
        {
          'spinner-small': isSmall,
          'absolute top-0 bottom-0 m-auto': isCentered
        },
        className
      )}
      data-testid="loading-spinner"
      role="img"
      aria-label={loadingText}
      {...props}
    />
  ) : (
    <div
      className={cx(
        'my-5',
        'text-cool-grey-600',
        'font-sans',
        'text-md',
        'font-bold',
        'tracking-widest',
        'text-center',
        'uppercase',
        'animate-pulse-fade',
        className
      )}
    >
      {loadingText}
    </div>
  );
