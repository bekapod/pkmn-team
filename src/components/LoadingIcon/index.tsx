import React from "react";
import styled from "styled-components/macro";
import { pulseFade } from "../../helpers/animations";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";

const Styled = styled.div`
  animation: ${pulseFade} 2s linear infinite;
  margin: ${variables.spacing.xl}px 0;
  color: ${variables.colors.grayDark};
  font-family: ${variables.fonts.base};
  font-size: ${variables.fontSizes.md}px;
  font-weight: 700;
  letter-spacing: ${variables.spacing.xxs}px;
  line-height: ${lineHeight("md")};
  text-align: center;
  text-transform: uppercase;
`;

const LoadingIcon = () => <Styled>Loading</Styled>;

export default LoadingIcon;
