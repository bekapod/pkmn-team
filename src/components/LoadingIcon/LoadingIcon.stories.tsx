import { Meta, Story } from '@storybook/react/types-6-0';
import { ComponentProps } from 'react';
import { LoadingIcon } from './LoadingIcon';

export default {
  title: 'Components/Loading Icon',
  component: LoadingIcon
} as Meta<ComponentProps<typeof LoadingIcon>>;

export const loadingIcon: Story<ComponentProps<typeof LoadingIcon>> = args => (
  <LoadingIcon {...args} />
);
