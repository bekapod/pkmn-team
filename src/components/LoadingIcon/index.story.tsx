// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import LoadingIcon from ".";

storiesOf("LoadingIcon", module).add("default", () => <LoadingIcon />);
