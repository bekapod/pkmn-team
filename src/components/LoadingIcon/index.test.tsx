import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import LoadingIcon from ".";

describe("<LoadingIcon />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<LoadingIcon />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
