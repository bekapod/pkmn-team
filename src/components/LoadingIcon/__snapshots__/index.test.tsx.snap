// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIcon /> renders without crashing 1`] = `
<div
  className="LoadingIcon__Styled-sc-1njdtjv-0 gEufFU"
>
  Loading
</div>
`;
